metrics/metrics.go

```diff
@@ -192,8 +192,8 @@ func (m *shadowsocksMetrics) AddClosedTCPConnection(clientLocation, accessKey, s
 	m.dataBytes.WithLabelValues("c<p", "tcp", clientLocation, status, accessKey).Add(float64(data.ProxyClient))
 }
 
-func (m *shadowsocksMetrics) AddTCPProbe(clientLocation, drainResult string, data ProxyMetrics) {
-	m.tcpProbes.WithLabelValues(clientLocation, drainResult).Observe(float64(data.ClientProxy))
+func (m *shadowsocksMetrics) AddTCPProbe(clientLocation, port, drainResult string, data ProxyMetrics) {
```

Comment:
Please make the port an integer. Convert to string in the WithLabelValues call.

Reply:
done

Reply:
Tested! Here's example output from /metrics:

```
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9000",le="0"} 1
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9000",le="48"} 2
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9000",le="49"} 2
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9000",le="50"} 2
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9000",le="51"} 2
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9000",le="52"} 2
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9000",le="+Inf"} 2
shadowsocks_tcp_probes_sum{error="timeout",location="",port="9000"} 6
shadowsocks_tcp_probes_count{error="timeout",location="",port="9000"} 2
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9001",le="0"} 1
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9001",le="48"} 1
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9001",le="49"} 1
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9001",le="50"} 1
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9001",le="51"} 1
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9001",le="52"} 1
shadowsocks_tcp_probes_bucket{error="timeout",location="",port="9001",le="+Inf"} 1
shadowsocks_tcp_probes_sum{error="timeout",location="",port="9001"} 0
shadowsocks_tcp_probes_count{error="timeout",location="",port="9001"} 1
```